space_packet_parser/common.py

```diff
@@ -247,3 +246,144 @@ class StrParameter(_Parameter, str):
 
 
 ParameterDataTypes = Union[BinaryParameter, BoolParameter, FloatParameter, IntParameter, StrParameter]
+
+
+class Packet(dict):
+    """Packet representing parsed data items.
+
+    Container that stores the binary packet data (bytes) as an instance attribute and the parsed
+    data items in a dictionary interface. A ``Packet`` generally begins as an empty dictionary that gets
+    filled as the packet is parsed. To access the raw bytes of the packet, use the ``Packet.binary_data`` attribute.
+
+    Parameters
+    ----------
+    *args : Mapping or Iterable
+        Initial items to store in the packet, passed to the dict() constructor.
+    binary_data : bytes, optional
+        The binary data for a single packet as a bytes object / subclass. This binary data is stored
+        and used for parsing the data items. Internally we are tracking the parsing position within
+        this binary_data object and trying to read specific bit ranges from it.
+    **kwargs : dict
+        Additional packet items to store, passed to the dict() constructor.
+    """
+    def __init__(self, *args, binary_data: bytes = b"", **kwargs):
+        if "raw_data" in kwargs:
+            warnings.warn("The 'raw_data' keyword argument is deprecated and will be removed in a future release. "
+                          "Use 'binary_data' instead.", DeprecationWarning, stacklevel=2)
+            binary_data = kwargs.pop("raw_data")
+        self.binary_data = binary_data
+        self._parsing_pos = 0
+        super().__init__(*args, **kwargs)
+
+    @property
+    def header(self) -> dict:
+        """The header content of the packet."""
+        warnings.warn("The header property is deprecated and will be removed in a future release. "
+                      "To access the header fields of a CCSDS packet, use the CCSDSPacketBytes class.",
```

Comment:
Is the intention that users should be creating `Packets` where the `binary_data` is a `CCSDSPacketBytes` object rather than a plain `bytes` object? Then they would call `my_packet.binary_data.header`? I think we should be more explicit about how to do this and maybe even explain that this is to support generic, non CCSDS packets. Also this applies to the warning in the `user_data` property.